File: jpi2/src/main/java/org/jenkinsci/gradle/plugins/jpi2/accmod/MavenLoggingBridge.java
package org.jenkinsci.gradle.plugins.jpi2.accmod;

import org.apache.maven.plugin.logging.Log;
import org.slf4j.Logger;
import org.slf4j.LoggerFactory;

/**
 * Adapts the Maven {@code Log} interface to SLF4J so that {@code kohsuke.accmod.Checker}
 * — which expects Maven logging — can write through Gradle's logging infrastructure.
 */
public class MavenLoggingBridge implements Log {
    private static final Logger LOGGER = LoggerFactory.getLogger(MavenLoggingBridge.class);
    private static final String EXCEPTION_MESSAGE = "An error occurred";

    @Override
    public boolean isDebugEnabled() {
        return LOGGER.isDebugEnabled();
    }

    @Override
    public void debug(CharSequence content) {
        LOGGER.debug(content == null ? null : content.toString());
    }

    @Override
    public void debug(CharSequence content, Throwable error) {
        LOGGER.debug(content == null ? null : content.toString(), error);
    }

    @Override
    public void debug(Throwable error) {
        LOGGER.debug(EXCEPTION_MESSAGE, error);
    }

    @Override
    public boolean isInfoEnabled() {
        return LOGGER.isInfoEnabled();
    }

    @Override
    public void info(CharSequence content) {
        LOGGER.info(content == null ? null : content.toString());
    }

    @Override
    public void info(CharSequence content, Throwable error) {
        LOGGER.info(content == null ? null : content.toString(), error);
    }

    @Override
    public void info(Throwable error) {
        LOGGER.info(EXCEPTION_MESSAGE, error);
    }

    @Override
    public boolean isWarnEnabled() {
        return LOGGER.isWarnEnabled();
    }

    @Override
    public void warn(CharSequence content) {
        LOGGER.warn(content == null ? null : content.toString());
    }

    @Override
    public void warn(CharSequence content, Throwable error) {
        LOGGER.warn(content == null ? null : content.toString(), error);
    }

    @Override
    public void warn(Throwable error) {
        LOGGER.warn(EXCEPTION_MESSAGE, error);
    }

    @Override
    public boolean isErrorEnabled() {
        return LOGGER.isErrorEnabled();
    }

    @Override
    public void error(CharSequence content) {
        LOGGER.error(content == null ? null : content.toString());
    }

    @Override
    public void error(CharSequence content, Throwable error) {
        LOGGER.error(content == null ? null : content.toString(), error);
    }

    @Override
    public void error(Throwable error) {
        LOGGER.error(EXCEPTION_MESSAGE, error);
    }
}

File: src/main/kotlin/org/jenkinsci/gradle/plugins/accmod/MavenLoggingBridge.kt
package org.jenkinsci.gradle.plugins.accmod

import org.apache.maven.plugin.logging.Log
import org.slf4j.Logger
import org.slf4j.LoggerFactory

object MavenLoggingBridge : Log {
    private val logger: Logger = LoggerFactory.getLogger(MavenLoggingBridge.javaClass)
    private const val exceptionMessage: String = "An error occurred"

    override fun isDebugEnabled(): Boolean = logger.isDebugEnabled

    override fun debug(content: CharSequence?) {
        logger.debug(content as String?)
    }

    override fun debug(content: CharSequence?, error: Throwable?) {
        logger.debug(content as String?, error)
    }

    override fun debug(error: Throwable?) {
        logger.debug(exceptionMessage, error)
    }

    override fun isInfoEnabled(): Boolean = logger.isInfoEnabled

    override fun info(content: CharSequence?) {
        logger.info(content as String?)
    }

    override fun info(content: CharSequence?, error: Throwable?) {
        logger.info(content as String?, error)
    }

    override fun info(error: Throwable?) {
        logger.info(exceptionMessage, error)
    }

    override fun isWarnEnabled(): Boolean = logger.isWarnEnabled

    override fun warn(content: CharSequence?) {
        logger.warn(content as String?)
    }

    override fun warn(content: CharSequence?, error: Throwable?) {
        logger.warn(content as String?, error)
    }

    override fun warn(error: Throwable?) {
        logger.warn(exceptionMessage, error)
    }

    override fun isErrorEnabled(): Boolean = logger.isErrorEnabled

    override fun error(content: CharSequence?) {
        logger.error(content as String?)
    }

    override fun error(content: CharSequence?, error: Throwable?) {
        logger.error(content as String?)
    }

    override fun error(error: Throwable?) {
        logger.error(exceptionMessage, error)
    }
}
